devito/types/dense.py

```diff
@@ -793,8 +793,8 @@ def _arg_defaults(self, alias=None):
         args = ReducerMap({key.name: self._data_buffer})
 
         # Collect default dimension arguments from all indices
-        for i, s in zip(self.dimensions, self.shape):
-            args.update(i._arg_defaults(_min=0, size=s))
+        for a, i, s in zip(key.dimensions, self.dimensions, self.shape):
```

Comment:
@FabioLuporini quite surprised this never broke anything

Reply:
I'm now confused. 

Shouldn't it be `key._data_buffer`?
Shouldn't it be `key.shape`?

Might just be early morning sleepy brain

Reply:
No:
- `key/alias` is the object and names used to build the operator, i.e the "key" of key-value pairs
- `self` is the runtime argument with the runtime shapes/values/data/....

So you want to return the {key: self} key-pair values

Reply:
ah yes you're right of course